src/workflows/services/common_service.py

```diff
@@ -240,6 +239,41 @@ def connect(self, frontend=None, commands=None):
         if commands:
             self.__pipe_commands = commands
 
+        if liveness := self._environment.get("liveness"):
+            from wsgiref.simple_server import make_server
+
+            def alive(environ, start_response):
+                self.__alive = False
+                self.__send_to_frontend({"band": "liveness_check"})
+
+                # timeout of liveness check, in seconds
+                timeout = time.time() + liveness.get("timeout", 30)
+                while True:
+                    if self.__alive or time.time() > timeout:
+                        break
+
+                if self.__alive:
+                    status = "200 OK"
```

Comment:
Is this something we want to report a prometheus metric on?

Reply:
Potentially interesting/useful, but perhaps something to revisit later if we think it's needed?

Reply:
If it's more complicated than just instantiating a single counter, then yeah probably not worth a lot of work. I thought the services were already instrumented.